### PR TITLE
Jetpack: add post-publish-qr-post-panel block editor plugin

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds the scaffolding for the QRPost panel plugin

Fixes https://github.com/Automattic/jetpack/issues/23275

#### Changes proposed in this Pull Request:

* Jetpack: add post-publish-qr-post-panel block editor plugin

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Enable [Beta extensions](https://github.com/Automattic/jetpack/tree/master/projects/plugins/jetpack/extensions#beta-extensions)
* Create a new post
* Click on `Publish` button
* The pre-publish panel will be open. Publish the post
* Confirm you see the QR Post panel there.

https://user-images.githubusercontent.com/77539/157131838-92f60dfd-cae5-4cf9-a0d4-044d4d197e60.mov

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/components/icons/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/components/icons/index.js:1]: Are there going to be more icons for this plugin? If not, maybe we can forgo the extra file and directory, and instead add the svg directly in `projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/index.js`?

File: projects/plugins/jetpack/class.jetpack-plan.php [hunk 2]
```diff
@@ -47,6 +47,7 @@ class Jetpack_Plan {
 				'whatsapp-button',
 				'social-previews',
 				'videopress',
+				'post-publish-qr-post-panel',
```
Comment (jeherve) [inline projects/plugins/jetpack/class.jetpack-plan.php:50]: I don't think we need this if the panel will be available to all sites on wpcom simple; we can bypass all kinds of plan checks if we know the feature will be free for everyone.

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/class.jetpack-plan.php (hunk 2 above)
Comment (retrofox) [reply projects/plugins/jetpack/class.jetpack-plan.php:50]: When removing the plugin is not loaded :-/

File: projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/components/icons/index.js (hunk 1 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/plugins/post-publish-qr-post-panel/components/icons/index.js]: addressed here https://github.com/Automattic/jetpack/pull/23281/commits/06e0388bef25b7afe0df2947b4ee46ffbc918e9a

**Review by oskosk — APPROVED** [review]
Works fine 👌

File: projects/plugins/jetpack/class.jetpack-plan.php (hunk 2 above)
Comment (oskosk) [reply projects/plugins/jetpack/class.jetpack-plan.php:50]: This looks tidy to me. Let's go like this and iterate later if there's a better alternative

Comment (retrofox) [reply projects/plugins/jetpack/class.jetpack-plan.php:50]: > Great news! One last step: head over to your WordPress.com diff, D76203-code, and commit it. Once you've done so, come back to this PR and add a comment with your changeset ID.

r241291-wpcom

Comment (jeherve) [reply projects/plugins/jetpack/class.jetpack-plan.php:50]: I opened #23290 to simplify things. As a result you shouldn't have to declare support on WordPress.com simple for the new feature.

**Pull request merged**